core-kotlin-modules/core-kotlin-collections-map/src/test/kotlin/com/baeldung/removeentry/RemoveMapEntryDuringIterationUnitTest.kt

```diff
@@ -0,0 +1,63 @@
+package com.baeldung.removeentry
+
+import org.junit.jupiter.api.Test
+import java.util.ConcurrentModificationException
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+
+class RemoveMapEntryDuringIterationUnitTest {
+
+    @Test
+    fun `test removing entry from map while iterating using for loop should remove successfully`() {
+        val mutableMap = mutableMapOf(1 to 3, 2 to 3, 3 to 2, 4 to 2, 7 to 2, 9 to 1)
+        val iterator = mutableMap.entries.iterator()
+
+        while (iterator.hasNext()) {
+            val entry = iterator.next()
+            if (entry.key % 2 == 0) {
+                iterator.remove()
+            }
+        }
+
+        assertEquals(4, mutableMap.size)
+    }
+
+
+    @Test
+    fun `test removing entry from map while iterating with for loop should fail with ConcurrentModificationException`() {
+        val mutableMap = mutableMapOf(1 to 3, 2 to 3, 3 to 2, 4 to 2, 7 to 2, 9 to 1)
+        assertFailsWith<ConcurrentModificationException> {
+            for ((key, value) in mutableMap) {
+                if (key % 2 == 0) {
+                    mutableMap.remove(key)
+                }
+            }
+        }
+    }
+
+    @Test
+    fun `test removing entry from map using predicate in removeIf method should remove successfully`() {
+        val mutableMap = mutableMapOf(1 to 3, 2 to 3, 3 to 2, 4 to 2, 7 to 2, 9 to 1)
+        val isEven: (Int) -> Boolean = { it -> it % 2 == 0 }
+        mutableMap.entries.removeIf { isEven(it.key) }
```

Comment:
I think the predicate example might be simpler to understand with one less layer of indirection:
```suggestion
    mutableMap.entries.removeIf { it.key % 2 == 0 }
```

Reply:
Done